test/integration/redirect_integration_test.cc

```diff
@@ -143,10 +161,16 @@ TEST_P(RedirectIntegrationTest, BasicInternalRedirect) {
   EXPECT_EQ("200", response->headers().getStatusValue());
   EXPECT_EQ(1, test_server_->counter("cluster.cluster_0.upstream_internal_redirect_succeeded_total")
                    ->value());
-  EXPECT_THAT(waitForAccessLog(access_log_name_), HasSubstr("internal_redirect"));
+  // 302 was never returned downstream
+  EXPECT_EQ(0, test_server_->counter("http.config_test.downstream_rq_3xx")->value());
```

Comment:
add:
EXPECT_EQ(1, test_server_->counter("http.config_test.downstream_rq_2xx")->value());

Reply:
Done, here and the other test case.